x/stake/endblock.go

```diff
@@ -252,3 +266,84 @@ func handleMatureUnbondingDelegations(k keeper.Keeper, ctx sdk.Context) ([]types
 
 	return completed, events
 }
+
+const (
+	maxProcessedRefundCount = 50
+)
+
+func handleRefundCrossStake(ctx sdk.Context, k keeper.Keeper) sdk.Events {
+	iterator := k.IteratorAllDelegations(ctx)
+	defer iterator.Close()
+	var refundEvents sdk.Events
+	count := 0
+	bcValidators := k.GetAllValidators(ctx)
```

Comment:
Is not it `bscValidators`, why bcValidators?

And also you did not inject the side chain id into the context, so it wont get the BSC delegation. 

Some example code:
```
	if sdk.IsUpgrade(sdk.BEP128) {
		sideChainIds, storePrefixes := k.ScKeeper.GetAllSideChainPrefixes(ctx)
		if len(sideChainIds) == len(storePrefixes) {
			for i := range storePrefixes {
				sideChainCtx := ctx.WithSideChainKeyPrefix(storePrefixes[i])
				csEvents = k.DistributeInBlock(sideChainCtx, sideChainIds[i])
			}
		} else {
			panic("sideChainIds does not equal to sideChainStores")
		}
	}
```